logstash-core/spec/logstash/agent_spec.rb

```diff
@@ -157,11 +157,17 @@
         subject.send(:reload_state!)
         expect(subject.pipelines[pipeline_id].config_str).to eq(new_config)
       end
+
       it "starts the pipeline" do
         expect(subject).to receive(:stop_pipeline)
         expect(subject).to receive(:start_pipeline)
         subject.send(:reload_state!)
       end
+
+      it "clears the metric collector" do
```

Comment:
Can we add a more concrete test where we send some events, make sure metric values are incremented and then send `reload_state`, then check if the values in metrics are zero?